### PR TITLE
DP-2366

### PR DESCRIPTION
**Pull request by amit-pio**
Introduced FollyConfigChecks.cmake.patch to handle ASan builds

### CODE REVIEW

**Review by mahaveerdarade — APPROVED** [review]
this patch file would be applied whenever user chose asan build.

File: CMake/FollyConfigChecks.cmake.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMake/FollyConfigChecks.cmake b/CMake/FollyConfigChecks.cmake
+index f93f4eac..290c49ec 100644
+--- a/CMake/FollyConfigChecks.cmake
++++ b/CMake/FollyConfigChecks.cmake
+@@ -31,6 +31,7 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
+     list(APPEND FOLLY_CXX_FLAGS -Wshadow-compatible-local)
+   endif()
+ 
++  list(APPEND FOLLY_CXX_FLAGS -DFOLLY_SANITIZE_ADDRESS=1)
+   CHECK_CXX_COMPILER_FLAG(-Wnoexcept-type COMPILER_HAS_W_NOEXCEPT_TYPE)
```
Comment (harishpujari) [inline CMake/FollyConfigChecks.cmake.patch:10]: FOLLY_SANITIZE_ADDRESS looks like being set for all cases, or is it conditional ?
Should be set only for ASAN build

File: CMake/FollyConfigChecks.cmake.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMake/FollyConfigChecks.cmake b/CMake/FollyConfigChecks.cmake
+index f93f4eac..290c49ec 100644
+--- a/CMake/FollyConfigChecks.cmake
++++ b/CMake/FollyConfigChecks.cmake
+@@ -31,6 +31,7 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
+     list(APPEND FOLLY_CXX_FLAGS -Wshadow-compatible-local)
```
Comment (harishpujari) [inline CMake/FollyConfigChecks.cmake.patch:6]: what is this for ?

File: CMake/FollyConfigChecks.cmake.patch (hunk 1 above)
Comment (amit-pio) [reply CMake/FollyConfigChecks.cmake.patch:10]: It is only set for ASAN builds and the handling is in hyc.py, tgt_checkout.py & tgt_build.py

**Review by harishpujari — APPROVED** [review]

**Pull request merged**